neptunecontrib/api/utils.py

```diff
@@ -272,47 +279,58 @@ def pickle_and_log_artifact(obj, filename, experiment=None):
         Create RandomForest object and log to Neptune::
 
             from sklearn.ensemble import RandomForestClassifier
-            from neptunecontrib.api import pickle_and_log_artifact
+            from neptunecontrib.api import log_pickle
 
-            with neptune.create_experiment():
-                rf = RandomForestClassifier()
-                pickle_and_log_artifact(rf, 'rf')
+            neptune.create_experiment()
+
+            rf = RandomForestClassifier()
+            log_pickle('rf.pkl', rf)
     """
     _exp = experiment if experiment else neptune
+    _exp.log_artifact(export_pickle(obj), filename)
 
-    with tempfile.TemporaryDirectory() as d:
-        filename = os.path.join(d, filename)
-        joblib.dump(obj, filename)
-        _exp.send_artifact(filename)
 
+def export_pickle(obj):
+    from io import BytesIO
+    import pickle
 
-def get_pickled_artifact(experiment, filename):
+    buffer = BytesIO()
+    pickle.dump(obj, buffer)
+    buffer.seek(0)
+
+    return buffer
+
+
+def get_pickle(filename, experiment):
     """Downloads pickled artifact object from Neptune and returns a Python object.
 
     Downloads the pickled object from artifacts of given experiment,
-     loads them and returns a Python object.
+     loads it to memory and returns a Python object.
 
     Args:
-        experiment(`neptune.experiments.Experiment`): Neptune experiment.
-        filename(str): filename under which object was saved in Neptune.
+        filename(str): filename under which object will be saved to Neptune.
+        experiment (:obj:`neptune.experiments.Experiment`, optional, default is ``None``):
+            | For advanced users only. Pass Neptune
+              `Experiment <https://docs.neptune.ai/neptune-client/docs/experiment.html#neptune.experiments.Experiment>`_
+              object if you want to control to which experiment data is logged.
+            | If ``None``, log to currently active, and most recent experiment.
```

Comment:
This description seems to be copied from log_ and doesn't quite fit :)

Reply:
changed to the old one -> I think this should probably be standardized across the lib